pymc3/distributions/timeseries.py

```diff
@@ -176,6 +184,76 @@ def logp(self, value):
         return at.sum(innov_like) + at.sum(init_like)
 
 
+class ARMARV(RandomVariable):
+    name = "ARMA"
+    ndim_supp = 0
+    ndims_params = [1, 1, 0, 0]
+    dtype = "floatX"
+    _print_name = ("ARMA", "\\operatorname{ARMA}")
+
+    def __call__(self, phi=[], theta=[], loc=0.0, scale=1.0, size=100, **kwargs) -> TensorVariable:
+        return super().__call__(phi, theta, loc, scale, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(
+            cls,
+            rng: np.random.default_rng(),
+            phi: np.ndarray,
+            theta: np.ndarray,
+            loc: float,
+            scale: float,
+            size: Tuple[int, ...],
+    ) -> np.ndarray:
+        phi = np.r_[1, -phi]
+        theta = np.r_[1, theta]
+        x = rng.normal(loc=loc, scale=scale, size=size)
+        return signal.lfilter(theta, phi, x, axis=0)
+
+
+ARMArv = ARMARV()
+
+class ARMA(Continuous):
+    rv_op = ARMArv
+
+    @classmethod
+    def dist(cls, order=None, phi=None, theta=None, size=100, **kwargs):
+        if phi is not None and theta is not None and order is not None:
+            raise ValueError('Only two of phi, theta and order are allowed')
+        if (phi is None or theta is None) and order is None:
+            raise ValueError('Specify both phi and theta, or specify order')
+        if phi is not None:
+            phi = at.as_tensor_variable(floatX(phi))
+        else:
+            phi = Normal.dist(0.0, 2.0, size=order[0])
+        if theta is not None:
+            theta = at.as_tensor_variable(floatX(theta))
+        else:
+            theta = Normal.dist(0.0, 2.0, size=order[1])
+        return super().dist([phi, theta], size=size, **kwargs)
+
+    def logp(Y, phi, theta, loc, scale):
+        T = Y.shape[0].eval()
+        P = phi.shape[0].eval()
+        Q = theta.shape[0].eval()
+        Y_lagged = shared(np.zeros((T,P)))
+        for i in range(P):
+            Y_lagged = at.set_subtensor(Y_lagged[(i+1):,i], Y[:-(i+1)])
+
+        mu = at.fill(Y, loc) + at.dot(Y_lagged, at.transpose(phi))
+        epsilon = at.sub(Y,mu)
+
+        epsilon_lagged = shared(np.zeros((T,Q)))
+        for j in range(Q):
+            epsilon_lagged = at.set_subtensor(epsilon_lagged[j:-1,j], epsilon[:-(j+1)])
```

Comment:
indexing here is wrong 

```suggestion
            epsilon_lagged = at.set_subtensor(epsilon_lagged[(j+1):,j], epsilon[:-(j+1)])
```